File: AtCoder/abc/162f.py
N = int(input())
A = list(map(int, input().split()))
INF = 10**18

dp0 = [-INF] * (N + 2)
dp1 = [-INF] * (N + 2)
dp2 = [-INF] * (N + 2)
dp0[0] = 0
dp0[1] = 0
dp1[0] = 0
dp1[1] = 0

for i, a in enumerate(A, start=2):
    dp0[i] = dp0[i - 2] + a
    dp1[i] = max(dp1[i - 2] + a, dp0[i - 1])
    dp2[i] = max(dp2[i - 2] + a, dp1[i - 1], dp0[i - 2])

if N % 2 == 1:
    print(dp2[-1])
else:
    print(max(dp1[-1], dp0[-1]))

File: AtCoder/abc/162f.rs
#![allow(unused_imports)]
#![allow(unused_macros)]
#![allow(non_snake_case)]
#![allow(dead_code)]
#![allow(unused_variables)]
#![allow(unused_mut)]
#![allow(unused_assignments)]
use proconio::input;
use proconio::marker::Usize1;
use std::collections::*;
use std::cmp::*;
use std::f64::consts::*;

const MOD: u64 = 1000000007;
const INF: i64 = std::i64::MAX / 4;

fn main() {
    input! {
        N: usize,
        A: [i64; N],
    }

    let (mut dp0, mut dp1, mut dp2) = (0, -INF, -INF);

    for (i, a) in A.iter().enumerate() {
        if i % 2 == 0 {
            dp1 = max(dp1, dp0);
            dp0 += a;
            dp2 += a;
        } else {
            dp2 = max(dp2, dp1);
            dp1 += a;
        }
    }

    if N % 2 == 0 {
        println!("{}", max(dp0, dp1));
    } else {
        println!("{}", max(dp1, dp2));
    }
}
